opba-banking-rest-api/src/main/java/de/adorsys/opba/tppbankingapi/ais/resource/TppBankingApiAisResource.java

```diff
@@ -11,15 +19,29 @@
 import de.adorsys.opba.tppbankingapi.ais.model.TransactionsResponse;
 
 @RestController
+@RequiredArgsConstructor
 public class TppBankingApiAisResource implements TppBankingApiAccountInformationServiceAisApi {
+    private final GetAccountsService accountsSerivce;
+    private final AccountListMapper accountListMapper;
+    private final PsuConsentSessionLoadingService psuConsentSessionLoadingService;
+    private final RedirectionService redirectionService;
 
     @Override
     public ResponseEntity<AccountList> getAccounts(String authorization, UUID xRequestID, String psuConsentSession) {
-        return new ResponseEntity<>(HttpStatus.NOT_IMPLEMENTED);
+        Optional<PsuConsentSession> existingPsuConsentSession = StringUtils.isNotBlank(psuConsentSession)
```

Comment:
Subjective and depends on the fact is it mock code or production code, but can be 
```
        return Optional.ofNullable("".equals(psuConsentSession) ? null : psuConsentSession)
                .flatMap(psuConsentSessionLoadingService::loadPsuConsentSessionById)
                .flatMap(accountsSerivce::getAccountsFor)
                .map(accountListMapper::toAccountList)
                .map(it -> ResponseEntity.ok().body(it))
                .orElseGet(() -> new ResponseEntity(redirectionService.redirectForAuthorisation(), HttpStatus.SEE_OTHER));
```
instead of tracing which optional is where

Reply:
Yes, you are right.

To me `flatMap`s drastically reduce code readability. Separation by if-else blocks brings to me more clarity what is done when.

But I can follow you and adjust it, if others also support it.
@max402, @electronicpeter, what do you think?